src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -333,26 +348,30 @@ internal SourceText TryReadFileContent(CommandLineSourceFile file, IList<Diagnos
                             if (regexes[sectionIndex].IsMatch(relativePath))
                             {
                                 var section = config.NamedSections[sectionIndex];
-                                addDiagnosticOptions(section, optionsBuilder, config.PathToFile);
+                                addOptions(section, treeOptionsBuilder, analyzerOptionsBuilder, config.PathToFile);
                             }
                         }
                     }
                 }
 
                 // Avoid extra allocations by using null.
-                allDiagnosticOptions.Add(optionsBuilder.Count > 0 ? optionsBuilder.ToImmutable() : null);
-                optionsBuilder.Clear();
+                allTreeOptions.Add(treeOptionsBuilder.Count > 0 ? treeOptionsBuilder.ToImmutable() : null);
+                allAnalyzerOptions.Add(analyzerOptionsBuilder.Count > 0 ? analyzerOptionsBuilder.ToImmutable() : null);
```

Comment:
seems more complex to have to store null and then check for it.  Since an empty dictionary is just a singleton... why not just always produce the dictionary and just store something that is never null?

Reply:
I don't see any change between using null and a singleton here. The downstream APIs already handle null. That's a part of the broader Compilation API contract.